prov/gni/include/gnix_nic.h

```diff
@@ -86,12 +88,19 @@ struct gnix_nic {
 	struct list_head tx_desc_free_list;
 	struct gnix_tx_descriptor *tx_desc_base;
 	atomic_t outstanding_fab_reqs_nic;
+	fastlock_t wq_lock;
```

Comment:
Can you document this, as well as the other new fields in the comments above?  Convert to doxygen-style if that's where we're headed.